### ISSUE TITLE
superfluous_disable_command bug

### ISSUE THREAD

**Issue by artsimonyan23** [realm/SwiftLint#5788]
superfluous_disable_command incorrectly runs all custom rules.

**Comment by SimplyDanny** [comment]
Please provide a concrete example.

**Comment by mildm8nnered** [comment]
I wish I'd never touched this.

I think there is a problem here, where disabled regions are nested.

Given the following config:

```
only_rules:
  - custom_rules
  - superfluous_disable_command

# Custom rules
custom_rules:
  fatal_error:
    name: Fatal Error
    excluded: "Tests/*"
    message: Prefer using `queuedFatalError` over `fatalError` to avoid leaking compiler host machine paths.
    regex: \bfatalError\b
    match_kinds:
      - identifier
    severity: error
  fattal_error:
    name: Fattal Error
    excluded: "Tests/*"
    message: Prefer using `queuedFatalError` over `fatalError` to avoid leaking compiler host machine paths.
    regex: \bfattalError\b
    match_kinds:
      - identifier
    severity: error    
```

and the following input:

```
// swiftlint:disable fatal_error
// swiftlint:disable:next fattal_error
fatalError("something")
// swiftlint:enable fatal_error
```

then

```
% swiftlint tt.swift --config tt.config
Linting Swift files at paths tt.swift
Linting 'tt.swift' (1/1)
/Users/martin.redington/Documents/Source/SwiftLint/tt.swift:1:33: warning: Superfluous Disable Command Violation: SwiftLint rule 'fatal_error' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
/Users/martin.redington/Documents/Source/SwiftLint/tt.swift:3:1: warning: Superfluous Disable Command Violation: SwiftLint rule 'fattal_error' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
/Users/martin.redington/Documents/Source/SwiftLint/tt.swift:3:9223372036854775807: warning: Superfluous Disable Command Violation: SwiftLint rule 'fatal_error' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)
Done linting! Found 3 violations, 0 serious in 1 file.
```

But `fatalError` should not be triggering `superfluous_disable_command` here at all.

I think this is being caused by the way regions are returned from `SwiftLintFile`'s `regions(restrictingRuleIdentifiers:)` method, and I'm hoping we can fix it just by making a change there - right now each new `swiftlint:` command seems to generate a region with all disabled identifiers, but we probably really want a region for each command.

**Comment by Patrick-Kladek** [comment]
Any update on this?

**Comment by mildm8nnered** [comment]
> Any update on this?

Yes - there's a PR at https://github.com/realm/SwiftLint/pull/5797, which is awaiting code review, which should hopefully fix the issue.

**Comment by mildm8nnered** [comment]
This should be resolved by #5797
